src/realm/object-store/object_store.cpp

```diff
@@ -492,25 +492,34 @@ bool ObjectStore::verify_valid_additive_changes(std::vector<SchemaChange> const&
     return verifier.other_changes || (verifier.index_changes && update_indexes);
 }
 
-void ObjectStore::verify_valid_external_changes(std::vector<SchemaChange> const& changes)
+void ObjectStore::verify_valid_external_changes(std::vector<SchemaChange> const& changes,
+                                                bool is_schema_additive)
 {
     using namespace schema_change;
     struct Verifier : SchemaDifferenceExplainer {
         using SchemaDifferenceExplainer::operator();
 
+        Verifier(bool additive_schema) : m_additive_schema(additive_schema)
+        {}
+        
         // Adding new things is fine
         void operator()(AddTable) {}
         void operator()(AddInitialProperties) {}
         void operator()(AddProperty) {}
         void operator()(AddIndex) {}
         void operator()(RemoveIndex) {}
+        void operator()(RemoveProperty prop) {
+            if(!m_additive_schema)
+                SchemaDifferenceExplainer::operator()(prop);
```

Comment:
This is the minimal change we require in order to allow the new logic for additive schemas. We can also implement a new verifier, but this will be the only difference between the one we have right now and the new one.

Reply:
This change does not make any sense to me.

Reply:
OK, then I can explain to you why this is causing a failure in our tests. 

In `Schema::copy_keys_from` we append to the `persisted_properties` the missing `properties` in case of additive schemas.  This happens during the `Realm::update_schema`

So when we then verify the schema, for example when we open a read transaction on the `before` realm instance after a client reset, we call.
`ObjectStore::verify_valid_external_changes`
which eventually call this:
`std::vector<SchemaChange> Schema::compare`
and then `zip_matching`. Eg:
`
// Modify columns
    zip_matching(target_schema, *this, [&]( ....
`

Obviously since we have added all the properties to the `target_schema`, the current schema (`this`) does not contain them, thus, correctly it is reported that a property has been removed from the current schema.

Another possible fix could be to append the missing property to the current schema, but this is pretty much the same of skipping the check for `RemovePropery` in case of additive schemas.

Reply:
This is just working around a bug by silencing the place that's correctly reporting that we're in an invalid state. What's happening in the client reset before case is that we're reading the schema from the _latest_ version of the Realm, ending that read transaction, then starting a new one at the frozen version, which is before that column was added. `verify_valid_external_changes` is then reporting that the Realm's schema is not compatible with the schema of the read transaction we're looking at. Skipping that validation doesn't solve the problem that the schema is incorrect.

Without the changes in this PR reading the schema from the latest version is fine for frozen Realms because that's always a superset of the schema for the frozen version, and we simply ignored any extra things that'd been added. Since we're no longer ignoring the extra things, we need to actually read the schema from the correct transaction version.

Reply:
Ok, I guess you are referring to this line in here: 

```
void Realm::read_schema_from_group_if_needed()
{
....
    auto current_version = transaction().get_version_of_current_transaction().version;
.....
}
```

Reply:
this does not longer apply.